src/tools/illink/src/ILLink.RoslynAnalyzer/DataFlow/LocalDataFlowVisitor.cs

```diff
@@ -421,6 +422,32 @@ public override TValue VisitExpressionStatement (IExpressionStatementOperation o
 		public override TValue VisitInvocation (IInvocationOperation operation, LocalDataFlowState<TValue, TValueLattice> state)
 			=> ProcessMethodCall (operation, operation.TargetMethod, operation.Instance, operation.Arguments, state);
 
+		public override TValue VisitDelegateCreation (IDelegateCreationOperation operation, LocalDataFlowState<TValue, TValueLattice> state)
+		{
+			if (operation.Target is IFlowAnonymousFunctionOperation lambda) {
+				VisitFlowAnonymousFunction (lambda, state);
+
+				// Instance of a lambda or local function should be the instance of the containing method.
+				// Don't need to track a dataflow value, since the delegate creation will warn if the
+				// lambda or local function has an annotated this parameter.
+				var instance = TopValue;
+				return HandleDelegateCreation (lambda.Symbol, instance, operation);
+			}
+
+			Debug.Assert (operation.Target is IMethodReferenceOperation);
+			if (operation.Target is not IMethodReferenceOperation methodReference)
+				return TopValue;
+
+			TValue instanceValue = Visit (methodReference.Instance, state);
```

Comment:
Will this actually report IL2026 if the target method has RUC on it?
We never visit the method itself, we only visit the instance.

But I guess the RUC analyzer will do that instead, right? (we should really merge the two into one... ;-))

Reply:
Nice :-)